relayer/processor/path_processor.go

```diff
@@ -452,12 +452,37 @@ ChannelHandshakeLoop:
 
 // assembleMsgUpdateClient uses the ChainProvider from both pathEnds to assemble the client update header
 // from the source and then assemble the update client message in the correct format for the destination.
-func (pp *PathProcessor) assembleMsgUpdateClient(src, dst *pathEndRuntime) (provider.RelayerMessage, error) {
-	// If the client state trusted height is not equal to the client trusted state height, we cannot send a MsgUpdateClient
-	// until another block is observed. But we can send the rest without the MsgUpdateClient.
+func (pp *PathProcessor) assembleMsgUpdateClient(ctx context.Context, src, dst *pathEndRuntime) (provider.RelayerMessage, error) {
```

Comment:
This method would likely read better with something like

```go
trustedConsensusHeight := dst.clientTrustedState.ClientState.ConsensusHeight
clientConsensusHeight := dst.clientState.ConsensusHeight
```

because it's difficult to skim through the method when you need to closely read those two distinct values that have very similar paths.